lua/neo-tree/sources/common/hijack_cursor.lua

```diff
@@ -0,0 +1,36 @@
+local events = require("neo-tree.events")
+local manager = require("neo-tree.sources.manager")
+local log = require("neo-tree.log")
+
+local M = {}
+
+local setup_for_module = function(module)
+  return function()
+    local state = manager.get_state(module)
```

Comment:
This will actually only give you the state of the sidebar for the current tab. There may also be state objects for other tabs along with many state objects for windows where position = "current".

There is a `manager._for_each_state()` function that you can use for this:
https://github.com/nvim-neo-tree/neo-tree.nvim/blob/8c89efb93b8383666a6f989762c2f610f5f2c930/lua/neo-tree/sources/manager.lua#L54

Reply:
Nice, this simplifies everything :)

Reply:
Although, this doesn't seem to work for some reason. Any idea what could that be?

Reply:
It seems that `all_states` is not populated in the `manager` at the time I initialize hijack.

Reply:
I'm sorry, I kind of mislead you here. This doesn't really work because states are created and destroyed as the user opens new trees in new tabs/windows and destroys those tabs/windows.

The event should just be subscribed once, and then every time it fires you should figure out which neo-tree window you are in (if any), then retrieve the state for that window or tab and move the cursor.

You can tell if a neo-tree window is focused by checking the buffer local variables. This code is an example of getting state for the current window:
https://github.com/nvim-neo-tree/neo-tree.nvim/blob/8c89efb93b8383666a6f989762c2f610f5f2c930/lua/neo-tree/setup/init.lua#L282-L287

If the position IS current, the get_state call needs to include the window id.

Reply:
Here's one example of subscribing to an event, there are actually a lot of subscriptions in the setup/init methods:

https://github.com/nvim-neo-tree/neo-tree.nvim/blob/7f6fa04dbd8e8c79d1af33bc90e856b65d8641da/lua/neo-tree/setup/init.lua#L704-L708

You can also view the help file for more details about the events and arguments:
```
:h neo-tree-events
```

Reply:
all clear now, thanks